MoreLinq/Slice.cs

```diff
@@ -45,16 +45,20 @@ public static IEnumerable<T> Slice<T>(this IEnumerable<T> sequence, int startInd
             if (startIndex < 0) throw new ArgumentOutOfRangeException(nameof(startIndex));
             if (count < 0) throw new ArgumentOutOfRangeException(nameof(count));
 
-            // optimization for anything implementing IList<T>
-            return !(sequence is IList<T> list)
-                 ? sequence.Skip(startIndex).Take(count)
-                 : _(count); IEnumerable<T> _(int countdown)
-                 {
-                     var listCount = list.Count;
-                     var index = startIndex;
-                     while (index < listCount && countdown-- > 0)
-                         yield return list[index++];
-                 }
+            return
+                sequence is IList<T> list ? SliceList(list.Count, i => list[i])
+#if IREADONLY
```

Comment:
With #379 merged into master and if you merge master with you PR then you can remove all conditional checks for `IREADONLY`.

Reply:
Sure. I'll remove them from this Slice, then make another one to remove them in general.